### PR TITLE
Shipping and payment charges

### PR DESCRIPTION
**Pull request by lambreva**

### CODE REVIEW

File: src/Pyrexx/ZUGFeRD/Model/AllowanceCharge.php [hunk 1]
```diff
@@ -27,6 +28,22 @@ class AllowanceCharge
...
+    /**
+     * @var string
+     * @JMS\Type("string")
+     * @JMS\XmlElement(cdata=false, namespace="urn:un:unece:uncefact:data:standard:ReusableAggregateBusinessInformationEntity:12")
+     * @JMS\SerializedName("Reason")
+     */
+    private $reason;
+
+    /**
+     * @var TradeTax
```
Comment (otzy) [inline src/Pyrexx/ZUGFeRD/Model/AllowanceCharge.php:40]: should not we introduce a new type CategoryTradeTax?
CategoryTradeTax has just three properties and if we use TradeTax user can be confused or accidentally set properties that should not be set.

File: src/Pyrexx/ZUGFeRD/Model/Trade/Settlement.php [hunk 2]
```diff
@@ -45,12 +46,19 @@ class Settlement
...
-    private $monetarySummation;
+    private $allowanceCharges = [];
+
+    /**
+     * @var SpecifiedLogisticsServiceCharge
+     * @Type("Pyrexx\ZUGFeRD\Model\Trade\SpecifiedLogisticsServiceCharge")
+     * @XmlElement(cdata=false, namespace="urn:un:unece:uncefact:data:standard:ReusableAggregateBusinessInformationEntity:12")
+     * @SerializedName("SpecifiedLogisticsServiceCharge")
+     */
+    private $logisticsService;
```
Comment (otzy) [inline src/Pyrexx/ZUGFeRD/Model/Trade/Settlement.php]: may be name the property logisticsServiceCharge for consitency? and getter and setter as well

File: src/Pyrexx/ZUGFeRD/Model/AllowanceCharge.php (hunk 1 above)
Comment (lambreva) [reply src/Pyrexx/ZUGFeRD/Model/AllowanceCharge.php:40]: Hi, according to the scheme **CategoryTradeTax** is the name of the xml element, but the type of the element is TradeTaxType, [link to the ZUGFeRD scheme](https://github.com/Pyrexx/zugferd-php/blob/master/src/Pyrexx/ZUGFeRD/Assets/Schema/ZUGFeRD1p0_urn_un_unece_uncefact_data_standard_ReusableAggregateBusinessInformationEntity_12.xsd#L231) These are all fields that TradeTaxType could use: [TradeTaxType children](https://github.com/Pyrexx/zugferd-php/blob/master/src/Pyrexx/ZUGFeRD/Assets/Schema/ZUGFeRD1p0_urn_un_unece_uncefact_data_standard_ReusableAggregateBusinessInformationEntity_12.xsd#L334) If you decide that a new model is a better way I will create it.

Comment (otzy) [reply src/Pyrexx/ZUGFeRD/Model/AllowanceCharge.php:40]: you are right it's TradeTaxType. I just checked that in https://konik.io/ZUGFeRD-Spezifikation/Das-ZUGFeRD-Format_1p0_technische_Dokumentation.pdf only three fields from the TradeTaxType are used when for the tax is inside SpecifiedTradeAllowanceCharge tag (page 31 and page 72)
But I agree, we don't need an extra type for that.

File: src/Pyrexx/ZUGFeRD/Model/Trade/Settlement.php (hunk 2 above)
Comment (lambreva) [reply src/Pyrexx/ZUGFeRD/Model/Trade/Settlement.php]: Hi, sorry for the delay, just renamed **logisticsService** property.

**Pull request merged**